### PR TITLE
Add support for onConnect in new interfaces

### PR DESCRIPTION
**Pull request by timleslie**

### CODE REVIEW

File: packages-next/types/src/index.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ export type KeystoneConfig = {
   db: {
     adapter: 'mongoose' | 'knex';
     url: string;
+    onConnect?: (keystone: Keystone) => any;
```
Comment (mitchellhamilton) [inline packages-next/types/src/index.ts:57]: `Keystone` refers to a different thing here which is not what the argument to onConnect is

**Review by nathsimpson — APPROVED** [review]

File: packages-next/types/src/index.ts (hunk 1 above)
Comment (timleslie) [reply packages-next/types/src/index.ts:57]: Oh, right... 🤦

Comment (timleslie) [reply packages-next/types/src/index.ts:57]: Can we maybe pick a different name for that thing then? Not for this PR obviously, just flagging it as a thing for the future.

**Review by mitchellhamilton — APPROVED** [review]

**Comment by gautamsi** [comment]
Can we make onConnect an array so that plugins in future can also be able to push to onConnect for their own keystone related code. Like extra seeding for the aux list or regular lists.

**Pull request merged**